doc: Update the bt_rpc doc with sysbuild (#16867)
nrfconnect/sdk-nrf: doc/nrf/libraries/bluetooth_services/rpc.rst

  * Bluetooth RPC Client and common software libraries that serialize the Bluetooth Host API and enable RPC communication.
    These libraries need to be part of the user application.

>   * :ref:`ipc_radio` application that includes the full configuaration for Bluetooth Low Energy stack, Bluetooth RPC Host, and common libraries that enable communication with the Bluetooth RPC Client.

Reviewer: ```suggestion
  * :ref:`ipc_radio` application that includes the full configuration of the Bluetooth Low Energy stack, Bluetooth RPC Host and common libraries that enable communication with the Bluetooth RPC Client.
```

  * :kconfig:option:`CONFIG_BT_DEVICE_NAME`
  * :kconfig:option:`CONFIG_NRF_RPC_CBKPROXY_OUT_SLOTS` on one core must be equal to :kconfig:option:`CONFIG_NRF_RPC_CBKPROXY_IN_SLOTS` on the other.

> To keep all the above configuration options in sync, create a snipptet with an overlay file that is shared between the application and network core.

Reviewer: ```suggestion
To keep all the above configuration options in sync, you can create a snippet with an overlay file that is shared between the application and network core.
```
Also consider linking to sysbuild snippet documentation: https://docs.nordicsemi.com/bundle/ncs-latest/page/zephyr/build/snippets/index.html.

Then, you can invoke build command like this:

.. parsed-literal::
   :class: highlight

>    west build -b *board_target* -Smy_snippet

Reviewer: ```suggestion
   west build -b *board_target* -S my_snippet
```

Configuration
*************

To enable the Bluetooth Low Energy RPC library, use the sysbuild configuration ``SB_CONFIG_NETCORE_IPC_RADIO`` along with  ``SB_CONFIG_NETCORE_IPC_RADIO_BT_RPC`` option.
Set the :makevar:`SNIPPET` to ``nordic-bt-rpc`` to apply necessary configuration for Bluetooth Low Energy RPC library on both cores.
If you use a custom board you need to create a custom snippet with similar configuration to the ``nordic-bt-rpc`` snippet.

Build the application using the following command:

.. code-block:: console

>    west build -b nrf5340dk/nrf5340/cpuapp -Snordic-bt-rpc -- -DSB_CONFIG_NETCORE_IPC_RADIO=y -DSB_CONFIG_NETCORE_IPC_RADIO_BT_RPC=y

Reviewer: ```suggestion
   west build -b board_name -S nordic-bt-rpc -- -DSB_CONFIG_NETCORE_IPC_RADIO=y -DSB_CONFIG_NETCORE_IPC_RADIO_BT_RPC=y
```

.. note::

>    The samples which support the Bluetooth Low Energy RPC use the :makevar:`FILE_SUFFIX` along with :makevar:`SNIPPET` to adjust the network/radio core selection and configuration.

Reviewer: Suggestion:
```suggestion
   The samples which support the Bluetooth Low Energy RPC use the :makevar:`FILE_SUFFIX` along with :makevar:`SNIPPET` to adjust the selection and configuration of the network/radio core firmware.
```

Samples using the library
*************************

The following |NCS| sample and application use this library:

* :ref:`ipc_radio`
* :ref:`ble_rpc_host`

The following |NCS| samples can optionally use this library:

* :ref:`central_uart`
* :ref:`peripheral_hids_mouse`
* :ref:`peripheral_uart`

> The :ref:`ipc_radio` sample, is a configurable sample, which exposes the Bluetooth LE stack functionality that runs on an MCU with radio (for example, the nRF5340 network core) to another CPU using the :ref:`nrfxlib:nrf_rpc`.

Reviewer: ```suggestion
The :ref:`ipc_radio` sample, is a configurable application, that exposes the Bluetooth LE stack functionality running on an MCU with radio (for example, the nRF5340 network core) to another CPU through the :ref:`nrfxlib:nrf_rpc`.
```

> When using the :ref:`ipc_radio` as Bluetooth Low Energy over RPC reconfigure it by setting the ``SB_CONFIG_NETCORE_IPC_RADIO_BT_RPC=y`` Sysbuild Kconfig option to run the Bluetooth LE stack on the network core.

Reviewer: ```suggestion
When using the :ref:`ipc_radio` as Bluetooth Low Energy over RPC reconfigure it by setting the ``SB_CONFIG_NETCORE_IPC_RADIO_BT_RPC=y`` Sysbuild Kconfig option to run the whole Bluetooth LE stack on the network core.
```

> When using the :ref:`ipc_radio` as Bluetooth Low Energy over RPC reconfigure it by setting the ``SB_CONFIG_NETCORE_IPC_RADIO_BT_RPC=y`` Sysbuild Kconfig option to run the Bluetooth LE stack on the network core.

Reviewer 2: ```suggestion
When using the :ref:`ipc_radio` as Bluetooth Low Energy over RPC, reconfigure it by setting the ``SB_CONFIG_NETCORE_IPC_RADIO_BT_RPC`` sysbuild Kconfig option to ``y`` to run the whole Bluetooth LE stack on the network core.
```

> For more details, see :ref:`ipc_radio_config`.

Reviewer 2: ```suggestion
For more details, see the :ref:`Configuration <ipc_radio_config>` section of the IPC radio firmware application guide.
```
